{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d3725635-4116-4179-aee6-a44e87a08029",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['https://github.com/google/deepvariant', 'https://github.com/ivanseidel/IAMDinosaur', 'https://github.com/broadinstitute/gatk', 'https://github.com/peteanderson80/bottom-up-attention', 'https://github.com/griffithlab/rnaseq_tutorial', 'https://github.com/primaryobjects/AI-Programmer', 'https://github.com/vgteam/vg', 'https://github.com/arq5x/bedtools2', 'https://github.com/kblin/ncbi-genome-download', 'https://github.com/marbl/CHM13', 'https://github.com/loganwright/Genome', 'https://github.com/tseemann/prokka', 'https://github.com/evo-design/evo', 'https://github.com/cmdcolin/awesome-genome-visualization', 'https://github.com/deeptools/pyGenomeTracks', 'https://github.com/tanghaibao/jcvi', 'https://github.com/ablab/spades', 'https://github.com/brentp/mosdepth', 'https://github.com/marbl/canu', 'https://github.com/voutcn/megahit', 'https://github.com/jerryji1993/DNABERT', 'https://github.com/eggnogdb/eggnog-mapper', 'https://github.com/rrwick/Unicycler', 'https://github.com/vcftools/vcftools', 'https://github.com/ComparativeGenomicsToolkit/cactus', 'https://github.com/davidliwei/awesome-CRISPR', 'https://github.com/GMOD/jbrowse', 'https://github.com/tseemann/snippy', 'https://github.com/Ecogenomics/GTDBTk', 'https://github.com/malonge/RagTag', 'https://github.com/cov-lineages/pangolin', 'https://github.com/oschwengers/bakta', 'https://github.com/marbl/Mash', 'https://github.com/genomejs/dna2json', 'https://github.com/nanopore-wgs-consortium/NA12878', 'https://github.com/bxlab/metaWRAP', 'https://github.com/ablab/quast', 'https://github.com/vanheeringen-lab/genomepy', 'https://github.com/bactopia/bactopia', 'https://github.com/openvax/pyensembl', 'https://github.com/ParBLiSS/FastANI', 'https://github.com/ranjaykrishna/visual_genome_python_driver', 'https://github.com/broadinstitute/pilon', 'https://github.com/Gaius-Augustus/BRAKER', 'https://github.com/Ecogenomics/CheckM', 'https://github.com/genetics-statistics/GEMMA', 'https://github.com/tariqdaouda/pyGeno', 'https://github.com/hall-lab/speedseq', 'https://github.com/higlass/higlass', 'https://github.com/lh3/miniprot', 'https://github.com/bcgsc/abyss', 'https://github.com/sanger-pathogens/Roary', 'https://github.com/daler/pybedtools', 'https://github.com/rrwick/Trycycler', 'https://github.com/nextgenusfs/funannotate', 'https://github.com/bernatgel/karyoploteR', 'https://github.com/philres/ngmlr', 'https://github.com/ncbi/pgap', 'https://github.com/genometools/genometools', 'https://github.com/NVIDIA-Genomics-Research/GenomeWorks', 'https://github.com/ebi-pf-team/interproscan', 'https://github.com/dzerbino/velvet', 'https://github.com/Gaius-Augustus/Augustus', 'https://github.com/schneebergerlab/plotsr', 'https://github.com/pinellolab/CRISPResso2', 'https://github.com/isovic/racon', 'https://github.com/MariaNattestad/Ribbon', 'https://github.com/xiaolei-lab/rMVP', 'https://github.com/sigven/pcgr', 'https://github.com/flowhub-team/WholeGenomeSequencing-WGS', 'https://github.com/Kinggerm/GetOrganelle', 'https://github.com/MrOlm/drep', 'https://github.com/lmdu/pyfastx', 'https://github.com/WrightonLabCSU/DRAM', 'https://github.com/schatzlab/genomescope', 'https://github.com/alekseyzimin/masurca', 'https://github.com/moshi4/pyGenomeViz', 'https://github.com/shilrley6/Faster-R-CNN-with-model-pretrained-on-Visual-Genome', 'https://github.com/dasmoth/dalliance', 'https://github.com/sanger-pathogens/Artemis', 'https://github.com/genome-in-a-bottle/giab_data_indexes', 'https://github.com/dpryan79/ChromosomeMappings', 'https://github.com/marbl/Winnowmap', 'https://github.com/sanger-pathogens/circlator', 'https://github.com/pshtif/Genome2D', 'https://github.com/KamilSJaron/smudgeplot', 'https://github.com/bcgsc/NanoSim', 'https://github.com/aquaskyline/SOAPdenovo2', 'https://github.com/labgem/PPanGGOLiN', 'https://github.com/ucscGenomeBrowser/kent', 'https://github.com/ropensci/biomartr', 'https://github.com/griffithlab/GenVisR', 'https://github.com/tseemann/shovill', 'https://github.com/nextstrain/nextclade', 'https://github.com/aws-samples/aws-batch-genomics', 'https://github.com/NBISweden/GAAS', 'https://github.com/dariober/ASCIIGenome', 'https://github.com/PowerGenome/PowerGenome', 'https://github.com/Nextomics/NextPolish', 'https://github.com/lbcb-sci/raven', 'https://github.com/volcano-sh/kubegene', 'https://github.com/nanoporetech/megalodon', 'https://github.com/lbcb-sci/racon', 'https://github.com/GMOD/jbrowse-components', 'https://github.com/broadinstitute/viral-ngs', 'https://github.com/hms-dbmi/hic-data-analysis-bootcamp', 'https://github.com/alimanfoo/pysamstats', 'https://github.com/VGP/vgp-assembly', 'https://github.com/buds-lab/the-building-data-genome-project', 'https://github.com/pangenome/odgi', 'https://github.com/MAGICS-LAB/DNABERT_2', 'https://github.com/DRL/blobtools', 'https://github.com/ekimb/rust-mdbg', 'https://github.com/shenwei356/kmcp', 'https://github.com/buds-lab/building-data-genome-project-2', 'https://github.com/rgcgithub/regenie', 'https://github.com/franciscozorrilla/metaGEM', 'https://github.com/marcologous/genome', 'https://github.com/genome-spy/genome-spy', 'https://github.com/tangerzhang/ALLHiC', 'https://github.com/rrwick/Long-read-assembler-comparison', 'https://github.com/baoe/AlignGraph', 'https://github.com/MGXlab/CAT_pack', 'https://github.com/huangnengCSU/compleasm', 'https://github.com/BGI-Qingdao/TGS-GapCloser', 'https://github.com/broadinstitute/ichorCNA', 'https://github.com/epam/NGB', 'https://github.com/nickjcroucher/gubbins', 'https://github.com/chklovski/CheckM2', 'https://github.com/refresh-bio/agc', 'https://github.com/jiabowang/GAPIT', 'https://github.com/popgenmethods/smcpp', 'https://github.com/jotech/gapseq', 'https://github.com/GreenleafLab/chromVAR', 'https://github.com/genomejs/gql', 'https://github.com/AdmiralenOla/Scoary', 'https://github.com/arq5x/bedtools', 'https://github.com/BoevaLab/FREEC', 'https://github.com/nspies/genomeview', 'https://github.com/cdanielmachado/carveme', 'https://github.com/Ensembl/WiggleTools', 'https://github.com/aehrc/VariantSpark', 'https://github.com/bluenote-1577/skani', 'https://github.com/baoxingsong/AnchorWave', 'https://github.com/brentp/bwa-meth', 'https://github.com/HICAI-ZJU/Scientific-LLM-Survey', 'https://github.com/barricklab/breseq', 'https://github.com/MariaNattestad/Assemblytics', 'https://github.com/kcleal/gw', 'https://github.com/brentp/cruzdb', 'https://github.com/medvedevgroup/SibeliaZ', 'https://github.com/pirovc/genome_updater', 'https://github.com/gabraham/flashpca', 'https://github.com/hillerlab/TOGA', 'https://github.com/cruizperez/MicrobeAnnotator', 'https://github.com/lucapinello/CRISPResso', 'https://github.com/neherlab/pan-genome-analysis', 'https://github.com/linnabrown/run_dbcan', 'https://github.com/virajbdeshpande/AmpliconArchitect', 'https://github.com/cbg-ethz/V-pipe', 'https://github.com/bknyaz/sgg', 'https://github.com/nf-core/eager', 'https://github.com/Gabaldonlab/redundans', 'https://github.com/bigdatagenomics/mango', 'https://github.com/opencobra/memote', 'https://github.com/GMOD/Apollo', 'https://github.com/althonos/pyrodigal', 'https://github.com/JingweiJ/ActionGenome', 'https://github.com/FunctionLab/ExPecto', 'https://github.com/marbl/parsnp', 'https://github.com/cslarsen/arv', 'https://github.com/DiltheyLab/HLA-LA', 'https://github.com/biobakery/phylophlan', 'https://github.com/XiaoleiLiuBio/G2P', 'https://github.com/jameshadfield/phandango', 'https://github.com/Gaius-Augustus/GALBA', 'https://github.com/mcveanlab/mccortex', 'https://github.com/linzhi2013/MitoZ', 'https://github.com/nanoporetech/pipeline-structural-variation', 'https://github.com/SysBioChalmers/DLKcat', 'https://github.com/ShujiaHuang/ilus', 'https://github.com/ramanathanlab/genslm', 'https://github.com/rrwick/Perfect-bacterial-genome-tutorial', 'https://github.com/phac-nml/staramr', 'https://github.com/COMBINE-lab/pufferfish', 'https://github.com/mentatpsi/OSGenome', 'https://github.com/morrislab/phylowgs', 'https://github.com/bluenote-1577/sylph', 'https://github.com/DECIPHER-genomics/Genoverse', 'https://github.com/CompSynBioLab-KoreaUniv/FunGAP', 'https://github.com/qiyunlab/HGTector', 'https://github.com/iqbal-lab-org/pandora', 'https://github.com/ewels/AWS-iGenomes', 'https://github.com/SunPengChuan/wgdi', 'https://github.com/rpetit3/dragonflye', 'https://github.com/xavierdidelot/ClonalFrameML', 'https://github.com/dnanexus-archive/parliament2', 'https://github.com/eead-csic-compbio/get_homologues', 'https://github.com/cschin/Peregrine', 'https://github.com/Godotcoffee/AlignGraph2']\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "from tqdm import tqdm\n",
    "\n",
    "def fetch_github_repos(search_term, token, max_repos=100):\n",
    "    headers = {'Authorization': f'token {token}'}\n",
    "    base_url = \"https://api.github.com/search/repositories\"\n",
    "    params = {\n",
    "        \"q\": search_term,\n",
    "        \"sort\": \"stars\",\n",
    "        \"order\": \"desc\"\n",
    "    }\n",
    "    repos_urls = []\n",
    "    page = 1\n",
    "\n",
    "    # Initialize tqdm progress bar\n",
    "    pbar = tqdm(total=max_repos, desc=\"Fetching URLs\", unit=\"url\")\n",
    "\n",
    "    while len(repos_urls) < max_repos:\n",
    "        params['page'] = page\n",
    "        response = requests.get(base_url, headers=headers, params=params)\n",
    "        if response.status_code != 200:\n",
    "            break\n",
    "        data = response.json()\n",
    "        for item in data.get('items', []):\n",
    "            repos_urls.append(item['html_url'])\n",
    "            pbar.update(1)\n",
    "            if len(repos_urls) >= max_repos:\n",
    "                break\n",
    "        page += 1\n",
    "    \n",
    "    pbar.close()\n",
    "    return repos_urls\n",
    "\n",
    "# Use this function like so:\n",
    "token = 'YOUR_GITHUB_TOKEN'  # Replace with your actual GitHub token\n",
    "urls = fetch_github_repos('genome', token, max_repos=200)\n",
    "print(f\"Number of URLs loaded: {len(urls)}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "174bf32e-b362-4c82-9784-97824cad1cbd",
   "metadata": {},
   "outputs": [],
   "source": [
    "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"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1eae1cd9-c51a-4501-a63b-36b7f4cbd38a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: tqdm in c:\\users\\mahnoor\\anaconda3\\envs\\env3\\lib\\site-packages (4.66.2)\n",
      "Requirement already satisfied: colorama in c:\\users\\mahnoor\\anaconda3\\envs\\env3\\lib\\site-packages (from tqdm) (0.4.6)\n"
     ]
    }
   ],
   "source": [
    "!pip install tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b60e213-ee67-4bfc-951c-9dfcae2c7d63",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
